src/Operators/finitedifference.jl

```diff
@@ -1320,6 +1320,204 @@ Base.@propagate_inbounds function stencil_right_boundary(
+- Mono4: Monotonicity preserving harmonic mean, this implies a strong monotonicity constraint (eq 4)
+- Mono5: Monotonicity preserving, with extrema bounded by the edge cells in the stencil (eq 5)
+
+The diffusion implied by these methods is proportional to the local upwind CFL number.
+The `mismatch` Δ𝜙 = 0 returns the first-order upwind method. Special cases (discussed in Lin et al (1994)) include setting the 𝜙_min = 0 or 𝜙_max = saturation mixing ratio for water vapor
+are not considered here in favour of the generalized local extrema in equation (5a, 5b).
+"""
+abstract type LimiterConstraint end
+struct AlgebraicMean <: LimiterConstraint end
+struct PosDef <: LimiterConstraint end
+struct Mono4 <: LimiterConstraint end
+struct Mono5 <: LimiterConstraint end
+
+LinVanLeerC2F(; kwargs...) = LinVanLeerC2F(NamedTuple(kwargs))
+
+return_eltype(::LinVanLeerC2F, V, A, dt) =
+    Geometry.Contravariant3Vector{eltype(eltype(V))}
+
+return_space(
+    ::LinVanLeerC2F,
+    velocity_space::AllFaceFiniteDifferenceSpace,
+    arg_space::AllCenterFiniteDifferenceSpace,
+    dt,
+) = velocity_space
+
+function compute_Δ𝛼_linvanleer(a⁻, a⁰, a⁺, v, dt, ::Mono5)
+    Δ𝜙_avg = ((a⁰ - a⁻) + (a⁺ - a⁰)) / 2
+    min𝜙 = min(a⁻, a⁰, a⁺)
+    max𝜙 = max(a⁻, a⁰, a⁺)
+    𝛼 = min(abs(Δ𝜙_avg), 2 * (a⁰ - min𝜙), 2 * (max𝜙 - a⁰))
+    Δ𝛼 = sign(Δ𝜙_avg) * 𝛼 * (1 - sign(v) * v * dt)
+end
+
+function compute_Δ𝛼_linvanleer(a⁻, a⁰, a⁺, v, dt, ::Mono4)
+    Δ𝜙_avg = ((a⁰ - a⁻) + (a⁺ - a⁰)) / 2
+    c = sign(v) * v * dt
+    if sign(a⁰ - a⁻) == sign(a⁺ - a⁰) && Δ𝜙_avg != eltype(v)(0)
+        return ((a⁰ - a⁻) * (a⁺ - a⁰)) / (Δ𝜙_avg) * (1 - c)
+    else
+        return eltype(v)(0)
+    end
+end
+
+function posdiff(x, y)
+    ifelse(x - y >= eltype(x)(0), x - y, eltype(x)(0))
+end
+function compute_Δ𝛼_linvanleer(a⁻, a⁰, a⁺, v, dt, ::PosDef)
+    Δ𝜙_avg = ((a⁰ - a⁻) + (a⁺ - a⁰)) / 2
+    min𝜙 = min(a⁻, a⁰, a⁺)
+    max𝜙 = max(a⁻, a⁰, a⁺)
+    return sign(Δ𝜙_avg) *
+           min(abs(Δ𝜙_avg), 2 * posdiff(a⁺, min𝜙), 2 * posdiff(max𝜙, a⁺)) *
+           (1 - sign(v) * v * dt)
+end
+
+function compute_Δ𝛼_linvanleer(a⁻, a⁰, a⁺, v, dt, ::AlgebraicMean)
+    return ((a⁰ - a⁻) + (a⁺ - a⁰)) / 2 * (1 - sign(v) * v * dt)
+end
+
+function slope_limited_product(v, a⁻, a⁻⁻, a⁺, a⁺⁺, dt, method)
+    # Following Lin et al. (1994)
+    # https://doi.org/10.1175/1520-0493(1994)122<1575:ACOTVL>2.0.CO;2
+    if v >= 0
+        # Eqn (2,5a,5b,5c)
+        Δ𝛼 = compute_Δ𝛼_linvanleer(a⁻⁻, a⁻, a⁺, v, dt, method)
+        return v ⊠ (a⁻ ⊞ RecursiveApply.rdiv(Δ𝛼, 2))
+    else
+        # Eqn (2,5a,5b,5c)
+        Δ𝛼 = compute_Δ𝛼_linvanleer(a⁻, a⁺, a⁺⁺, v, dt, method)
+        return v ⊠ (a⁺ ⊟ RecursiveApply.rdiv(Δ𝛼, 2))
+    end
+end
+
+stencil_interior_width(::LinVanLeerC2F, velocity, arg, dt) =
+    ((0, 0), (-half - 1, half + 1), (0, 0))
+
+Base.@propagate_inbounds function stencil_interior(
+    ℱ::LinVanLeerC2F,
```

Comment:
I think we've decided to avoid script characters like `ℱ`. Maybe try sticking to ASCII, Greek, and super/subscripts to be consistent with the rest of our code.